Code/Framework/AzCore/AzCore/Component/ComponentApplication.h

```diff
@@ -392,5 +402,19 @@ namespace AZ
         //! @return true to indicate a record event operation should occur in the current Tick() call
         using RecordMetricsCallback = AZStd::function<bool(AZStd::chrono::steady_clock::time_point)>;
         RecordMetricsCallback m_recordMetricsOnTickCallback;
+
+    private:
+        enum class DevelopmentSettingsOverrides
+        {
+            None,                       // 0 = no overrides are allowed
+            CommandLineOnly,            // 1 = registry overrides are allowed from the command line
+            CommandLineAndProject,      // 2 = registry overrides are allowed from the command line, engine, gem, and project files
+            CommandLineProjectAndUser   // 3 = registry overrides are allowed from the command line, engine, gem, project, and user files
+        };
+
+        //! Determines which development settings (user registry files, project registry files, etc) should
+        //! be allowed to override the default settings.
+        //! @return the level of development settings overrides that are allowed
+        static constexpr DevelopmentSettingsOverrides GetDevelopmentSettingsOverrides();
```

Comment:
We can move the enum to be defined in the `ComponentApplication.cpp` as well as remove the declaration of `GetDevelopmentSettingsOverrides` in the class declaration.

Since we want `GetDevelopmentSettingsOverrides` to be constexpr it would make sense to be only inline in the cpp file